elements/lisk-transactions/src/10_delegate_transaction.ts

```diff
@@ -92,10 +100,27 @@ export class DelegateTransaction extends BaseTransaction {
 		const errors: TransactionError[] = [];
 		const sender = await store.account.get(this.senderId);
 
-		const usernameExists = store.account.find(
-			(account: Account) => account.username === this.asset.username,
+		// Data format for the registered delegates
+		// chain:usernames => { registeredDelegates: { username, address }[] }
+		const usernamesStr = await store.chain.get(DELEGATE_USERNAME_CHAIN_KEY);
+		const usernames = usernamesStr
+			? (JSON.parse(usernamesStr) as ChainUsernames)
+			: { registeredDelegates: [] };
+		const usernameExists = usernames.registeredDelegates.find(
+			delegate => delegate.username === this.asset.username,
```

Comment:
also here, try/catch

Reply:
I don't understand why we need a try/catch for a simple searcing in the array. This code is just checking if a particular array element exists. The finder function itself doesn't throw an error if the object not found.

Reply:
Oh I was referring to `JSON.parse`, not the find function

Reply:
I folllowe the pattern we used on other places as well: 

https://github.com/LiskHQ/lisk-sdk/blob/359700b09e0305f811e007b9281e7ff76ed0a23f/elements/lisk-dpos/src/delegates_list.ts#L68

https://github.com/LiskHQ/lisk-sdk/blob/359700b09e0305f811e007b9281e7ff76ed0a23f/elements/lisk-dpos/src/delegates_list.ts#L89

These JSON objects are not user-provided, so we trust that what we generated previously will be parsed. And in general, we rely on top-level code (framework) above libraries that will catch any error if occurred and will log.